### DISCUSSION TITLE
Time & data

### DISCUSSION THREAD

**Discussion by Trzinka** [ExperienceLovelace/ha-floorplan#430] [Q&A] [answered]
**Please help with the floor plan card! I think this is incorrect** `service_data: $(entity.state) `
```yaml
- element: sensor.time
  state_action:
    - service: floorplan.text_set
      service_data: $(entity.state)
    - service: floorplan.class_set
      service_data:
        class: static-temp
  tap_action: false
```

**Comment by exetico** [comment]
Hi @Trzinka Could you please consider looking for the answer in the docs, examples and YouTube videos?
I'm not here trying to be rude or anything, but I'm seeing many many questions from you, most of them should be covered in examples, previous questions, videos or related thinks. I hope you understand. I mean, it's great to have a dedicated user on board, but, I don't really think you're searching that much for each answer, before you're activating our good man @OzGav :) 

I also noticed that the translation does bring wierd formatted text. I'd suggest you to use a bit more time on that part. Like, there's no need for `##` headers, only bold text and things like that. I think I've tried to point that out in previous discussions.

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/f7d03609-d6cd-4e06-9125-68cf75d207c0)
_Jumping text, lots of sizes._

I like that the YAML code are formatted, though.

I hope you understand that we have plenty of resources out there, and we're spend out time to bring a lot of cases to the table. So, please, watch the videos, and play around with more of the examples. I hope you understand :+1: :smile: After you've looked at the docs, take a look at my collection for the YouTube videos here: https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/tree/8287cb442d98346f6e78400a4e9bdadafb995740

Personally I've deleted every email I get about topics from you. And I'm **not** trying to be rude or anything. Not at all. Personally I'm normally saving every ha-floorplan related discussions and issues, cause I like to find a day once in a while, finding time to reply, and give better answer (than a single line of code, or similar). But that high volume for a single user are simply a bit too overwhelming. And, now that I have found a almost 1:1 question-match with a video tutorial I made, I think you should take your time, watch the videos I've made to help the community, check all our examples, and prepare yourself for a few more technical steps. At that time, I'm sure other people are here to help! :smiley: 

And, going back to the original question:
I'm not 100% sure about what you need help to do? Think about the https://xyproblem.info/, if you need to ask about technical problems. That's helpful for us, and will 100% save us some time, trying to understand what you're trying to achieve.
We have plenty of information on how to use `text_set` with `entity.state` as the data for a svg text-element. I even have a example on how you can work with the entity-state, to manipulate and set the value to a svg text element (including date-formatting):
https://www.youtube.com/watch?v=GHcgwW-OOo8

There's also code examples attached to the video:
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/tree/main/6_Format-time_date.time-entity-from-military-to-AM_PM-in-state_action

Take a look at:
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/blob/8287cb442d98346f6e78400a4e9bdadafb995740/6_Format-time_date.time-entity-from-military-to-AM_PM-in-state_action/case3-final.yaml#L49

**Reply by Trzinka** [reply]
No offense! If you may have noticed, I solved most of the things myself, I share my knowledge with others, and so does the code. Admittedly, I should have thought about whether I had enough knowledge to tackle the floor plan first! Before I came across your instructions, I watched the video https://youtu.be/oJ9Yr2dSqUk?list=PLNaeA0iHcKdQIfb0TjdS93yqvXJHjTb45, which really attracted me because of the beautiful design (the video is very complicated and I had to watch it several times). at least roughly understand what and how). I was immediately put off by your video because you didn't use pictures of the floor plan (I didn't know at the time that I needed to thank you for the wonderful job of making the floor plan card). I also had no experience with Github unless I watched a video that explained the process step by step. @OzGav is a really nice guy like people on Discord or Facebook. As far as text formatting is concerned, there is indeed a gap between those of us who are advocates of landscape screens even when using a mobile phone for viewing. I also apologize that you feel the need to answer every question personally. I imagined that the discussion was intended to answer everyone and not just the "initiated". Regarding my messages you receive by email, the solution is simple, add me to spam. People's biggest curse is ignorance and to be honest I have been working with floor plan card for over 6 months now (total: making floor plan, svg file etc.) and as I wrote before, I may have gone overboard and should have used a map with pictorial elements!!! I didn't mean anything bad by what I wrote, all I can say is that you have done an outstanding job that unfortunately too many people don't have enough knowledge to use the floor plan card including myself. I wish you all the best.

**Comment by Trzinka** [comment, marked as answer]
# The solution is:

```yaml
      - entities:
          - sensor.time
          - sensor.date_long
        state_action:
          service: floorplan.text_set
          service_data: |
            > return `${entity.state}`;
```
